Fail to build due to pthread linking issue on Autoware 1.14.0

## Bug report

The version 1.14.0 of autoware fails to build due a linking failure to the pthread library. 

### Required information:

- Operating system and version: Ubuntu 18.04 x64
- Autoware installation type: Source
- Autoware version or commit hash: 1.14.0
- ROS distribution and version: melodic
- ROS installation type: deb
- Package or library, if applicable: ros_observer

### Description of the bug

```
Starting  >>> ros_observer                                                                                                                                                                                 
___________________________________________________________________________________________________________________________________________________________________________________________________________
Errors     << ros_observer:make /home/alexis/auto-ws/logs/ros_observer/build.make.005.log                                                                                                                  
/usr/bin/ld: CMakeFiles/ros_observer.dir/src/ros_observer.cpp.o: undefined reference to symbol 'pthread_mutexattr_settype@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [/home/alexis/auto-ws/devel/.private/ros_observer/lib/ros_observer/ros_observer] Error 1
make[1]: *** [CMakeFiles/ros_observer.dir/all] Error 2
make: *** [all] Error 2
cd /home/alexis/auto-ws/build/ros_observer; catkin build --get-env ros_observer | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
...........................................................................................................................................................................................................
Failed     << ros_observer:make           [ Exited with code 2 ]                                                                                                                                           
Failed    <<< ros_observer                [ 0.6 seconds ]                                                                                                                                                  
```


### Steps to reproduce the bug

```bash
git clone https://github.com/Autoware-AI/autoware.ai -b 1.14.0 --depth 1
cd autoware.ai
vcs import < autoware.ai.repos
cd ../..
rosdep update
rosdep install -y --from-paths src/autoware.ai/ --ignore-src --rosdistro melodic
catkin build ros_observer
```

### Expected behavior

It builds successfully

### Actual behavior

It doesn't build.

--- a/ros_observer/CMakeLists.txt
+++ b/ros_observer/CMakeLists.txt
@@ -34,6 +34,7 @@ add_executable(
 target_link_libraries(
   ros_observer
   rt ${Boost_LIBRARIES}
+  pthread
 )
 
 add_dependencies(

